### DISCUSSION TITLE
hydration is not work as doc describe

### DISCUSSION THREAD

**Discussion by TpTTT-T** [vercel/next.js#35810] [Help] [answered]
https://nextjs.org/learn/basics/data-fetching/pre-rendering

In this pre-rendering doc, 

> When a page is loaded by the browser, its JavaScript code runs and makes the page fully interactive.

> If your app has ineractive compnents like `<Link />,` they'll be active after JS loads

I have try this, but <Link /> still work even if javascript disable in browser

```
import Link from 'next/link'

function Router() {
    return (
        <div>
            <ul>
                <li>
                    <Link href="/">a</Link>
                </li>
                <li>
                    <Link href="/b">b</Link>
                </li>
            </ul>
        </div>
    )
}

export default Router
```

This is page on browser, I disable javascript in dev-tools and then refresh, when I click link a or link b, it still go to right page.

but doc describe that it only work after JS load .

**Comment by icyJoseph** [comment, marked as answer]
Hi,

First of all, I'd say it's best to wrap an anchor tag:

```jsx
<Link href="/my-page">
   <a>My Link</a>
</Link>
```

Second, if you analyse the DOM, even with JS disabled, you'll see than an actual `a` tag with `href` is rendered. 

What does this imply?
- Natural browser behaviour, that is, clicking on a link, takes you to the page.
- Next.js has the opportunity to do an "SPA" like transition

What does this imply, if JS is disabled?
- Natural browser behaviour, that is, clicking on a link, takes you to the page.
- Next.js doesn't have the opportunity to an "SPA" like transition

This means, that the entire page is destroyed, and navigating to the clicked page is an entirely new request.

To convince yourself of this, open the `devTools`, and in `console`, select `preserve logs`, and see that with JS disabled, clicking on the link logs:

> Navigated to http://localhost:3001/my-page

But, with JS enabled, nothing is printed, but the page "transitions" anyway. That's the SPA like behaviour of Next.js, that it can't do, if JS is disabled/blocked.

To sum up:

- Clicking `a` tags will always work (as long as there's an `href`, because that's just how browsers work)
- Interactivity depends on JS (SPA like page transitions, etc.)

## What the docs say:

It is important to carefully read every sentence in the docs:

> - Enable JavaScript in your browser and [check out this page](https://create-react-template.vercel.app/). This is a plain React.js app > built with [Create React App](https://create-react-app.dev/).
> - Now, disable JavaScript and access [the same page](https://create-react-template.vercel.app/) again.
> - You won’t see the app anymore — instead, it’ll say “You need to enable JavaScript to run this app.” This is because the app is not > pre-rendered into static HTML.

This part is actually, making the same point I am making. A `Create React App` site, with JS disabled, won't navigate! I think you must have missed that part. The docs are saying that, the benefit of Next.js is that even with JS disabled, things will work (for the most part)! 🎉 🎉

**Reply by TpTTT-T** [reply]
Really good explanation! You teach me more concept about how next js work on browser
